awx_collection/plugins/modules/tower_job_template.py

```diff
@@ -12,409 +13,562 @@
+        new_fields['project'] = project_id
+    if playbook is not None:
+        new_fields['playbook'] = playbook
+    if scm_branch is not None:
+        new_fields['scm_branch'] = scm_branch
+    if forks is not None:
+        new_fields['forks'] = forks
+    if limit is not None:
+        new_fields['limit'] = limit
+    if verbosity is not None:
+        new_fields['verbosity'] = verbosity
+    if extra_vars is not None:
+        new_fields['extra_vars'] = json.dumps(extra_vars)
+    if job_tags is not None:
+        new_fields['job_tags'] = job_tags
+    if force_handlers is not None:
+        new_fields['force_handlers'] = force_handlers
+    if skip_tags is not None:
+        new_fields['skip_tags'] = skip_tags
+    if start_at_task is not None:
+        new_fields['start_at_task'] = start_at_task
+    if timeout is not None:
+        new_fields['timeout'] = timeout
+    if use_fact_cache is not None:
+        new_fields['use_fact_cache'] = use_fact_cache
+    if host_config_key is not None:
+        new_fields['host_config_key'] = host_config_key
+    if ask_scm_branch_on_launch is not None:
+        new_fields['ask_scm_branch_on_launch'] = ask_scm_branch_on_launch
+    if ask_diff_mode_on_launch is not None:
+        new_fields['ask_diff_mode_on_launch'] = ask_diff_mode_on_launch
+    if ask_variables_on_launch is not None:
+        new_fields['ask_variables_on_launch'] = ask_variables_on_launch
+    if ask_limit_on_launch is not None:
+        new_fields['ask_limit_on_launch'] = ask_limit_on_launch
+    if ask_tags_on_launch is not None:
+        new_fields['ask_tags_on_launch'] = ask_tags_on_launch
+    if ask_skip_tags_on_launch is not None:
+        new_fields['ask_skip_tags_on_launch'] = ask_skip_tags_on_launch
+    if ask_job_type_on_launch is not None:
+        new_fields['ask_job_type_on_launch'] = ask_job_type_on_launch
+    if ask_verbosity_on_launch is not None:
+        new_fields['ask_verbosity_on_launch'] = ask_verbosity_on_launch
+    if ask_inventory_on_launch is not None:
+        new_fields['ask_inventory_on_launch'] = ask_inventory_on_launch
+    if ask_credential_on_launch is not None:
+        new_fields['ask_credential_on_launch'] = ask_credential_on_launch
+    if survey_enabled is not None:
+        new_fields['survey_enabled'] = survey_enabled
+    if become_enabled is not None:
+        new_fields['become_enabled'] = become_enabled
+    if diff_mode is not None:
+        new_fields['diff_mode'] = diff_mode
+    if allow_simultaneous is not None:
+        new_fields['allow_simultaneous'] = allow_simultaneous
+    if custom_virtualenv is not None:
+        new_fields['custom_virtualenv'] = custom_virtualenv
+    if job_slice_count is not None:
+        new_fields['job_slice_count'] = job_slice_count
+    if webhook_service is not None:
+        new_fields['webhook_service'] = webhook_service
+    if webhook_credential is not None:
+        new_fields['webhook_credential'] = webhook_credential_id
```

Comment:
This is a lot of additional code. The old module did an outright `params.copy()`, so there was no enumeration of the field names at all, and with this PR we're losing that, and we're not getting anything for it. As a _tiny_ improvement I've listed the params in other modules to keep it slightly more brief. In this case I think the ideal solution would be to remove params that shouldn't be treated in the standard way.